server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -322,6 +324,19 @@ protected BytesRef indexedValueForSearch(Object value) {
             return getTextSearchInfo().getSearchAnalyzer().normalize(name(), value.toString());
         }
 
+        /**
+         * Wildcard queries on keyword fields use the normalizer of the underlying field, regardless of their case sensitivity option
+         */
+        @Override
+        public Query wildcardQuery(
+            String value,
+            MultiTermQuery.RewriteMethod method,
+            boolean caseInsensitive,
+            SearchExecutionContext context
+        ) {
+            return super.wildcardQuery(value, method, caseInsensitive, true, context);
```

Comment:
Maybe calling `normalizedWildcardQuery` here would make the intent more obvious plus help any tracing back of where we make use of normalized wildcard queries

Reply:
Good idea, but I left the `caseInsensitive` parameter out of the `normalizedWildcardQuery` signature on purpose, so I need to call the protected method that takes both arguments here. My thinking was that we use `normalizedWildcardQuery` only from QueryStringQueryParser where we don't have the `caseInsensitive` option. Maybe you see a different solution?

Reply:
Ah ok. That makes sense.